ttl: allow outbound foreign keys on TTL tables

ecwall opened the pull request:
Fixes #101225

This change allows TTL tables to have outbound foreign keys. Outbound foreign
keys are unaffected by rows being deleted from the TTL table itself.

Release note (sql change): Allow outbound foreign keys on TTL tables.

ecwall committed:
ttl: remove unrelated columns from row_level_ttl

Removes a text column a few subtests have that is not part of the actual tests.

Release note: None

ecwall committed:
ttl: finish adding subtests to row_level_ttl

The row_level_ttl logic test is now broken down into subtests to more easily
isolate failures.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/101236)
<!-- Reviewable:end -->

ecwall committed:
ttl: allow outbound foreign keys on TTL tables

Fixes #101225

This change allows TTL tables to have outbound foreign keys. Outbound foreign
keys are unaffected by rows being deleted from the TTL table itself.

Release note (sql change): Allow outbound foreign keys on TTL tables.

otan commented on pkg/sql/catalog/tabledesc/validate.go:
```diff
@@ -198,23 +198,14 @@ func (desc *wrapper) ValidateForwardReferences(
-				desc.GetName(),
-			))
-		}
+	if desc.HasRowLevelTTL() && len(desc.InboundForeignKeys()) > 0 {
```
any reason not to support all the cases if we're not doing strict filtering?

ecwall replied:
I will handle it in a followup PR.

There is an issue with job SessionData not respecting `optDrivenFKCascadesClusterLimit` that I have to resolve https://github.com/cockroachdb/cockroach/issues/101265.

otan reviewed (approved):
probably want rafi to double check

rafiss reviewed (approved):
nice!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/101236#-:-NSq9W6sEv7dIUcs0HMO:b-7m74ox)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)



<!-- Sent from Reviewable.io -->

ecwall commented:
bors r=otan,rafiss
